// app/prepareAppContainer.js
var $ = require('jquery');

module.exports = function () {

    var $appContainer = $('<div id="app">').css({position: 'relative'});

    $(function () {
        $('body').append($appContainer);
    });

    var buses = [1, -10, 1, 1, 1, -10];

    buses.forEach(function (prize, index) {
        setTimeout(function () {
            var $bus = $('<div>');
            $bus.css({
                background: prize === -10 ? '#c00' : '#00c',
                width: 100,
                height: 50,
                position: 'absolute',
                top: 0,
                left: 0
            });

            $appContainer.append(
                $bus.animate(
                    {
                        left:200
                    },
                    4000,
                    'linear',
                    function () {
                        setTimeout(function () {
                            $bus.animate({
                                left: 400
                            }, 4000, 'linear', function () {
                                $(this).hide(300);
                            });
                        }, 1000)
                    }
                )
            );

            console.log(prize);
        }, 2000 * index + Math.random() * 500)
    });
};
